terratorch/tasks/segmentation_tasks.py

```diff
@@ -240,7 +242,7 @@ def training_step(self, batch: Any, batch_idx: int, dataloader_idx: int = 0) ->
 
         model_output: ModelOutput = self(x, **rest)
         loss = self.train_loss_handler.compute_loss(model_output, y, self.criterion, self.aux_loss)
-        self.train_loss_handler.log_loss(self.log, loss_dict=loss, batch_size=x.shape[0])
+        self.train_loss_handler.log_loss(self.log, loss_dict=loss, batch_size=y.shape[0])
```

Comment:
@blumenstiel can you please explain why you've changed from x to y?

Reply:
Because with the new multi-modal dataset x is a dict in order to pass all modalities to the model and does not have `.shape`.